astropy/table/table.py

```diff
@@ -2005,6 +2005,12 @@ def pformat(
 
         return lines
 
+    @deprecated(
+        since="7.0",
+        message="We are planning on deprecating Table.pformat_all",
```

Comment:
the phrasing I used is borrowed from the only other pending deprecation (`Table.show_in_browser`). I'm also not sure it's worth doing a *pending* deprecation if we're 100% sure that it will be deprecated; why not just deprecate it now then ?